### Exercício 1

Nesse primeiro exercício, vamos importar e transformar uma tabela do **conjunto de dados públicos do BigQuery**. Antes de partir para o código, você precisa ativá-los dentro da sua conta do BigQuery seguindo as instruções [desse guia](https://cloud.google.com/bigquery/public-data). Esse passo é importante para 1) liberar os datasets no BigQuery; e 2) conectar a eles usando o R usando o pacote `bigrquery`, se aplicável. Fique tranquilo(a), pois usar esses dados não gera cobrança na sua conta.

1. Nos **conjuntos de dados públicos do BigQuery**, dentro do dataset **[fda_food](https://console.cloud.google.com/bigquery?ws=!1m5!1m4!4m3!1sbigquery-public-data!2sfda_food!3sfood_events&pli=1&project=greenpeace-br-sp-01)**, faça a importação da tabela **food_events** para o R. Essa tabela reúne mais de 90 mil observações com reações adversas a alimentos. Mostre que a tabela foi importada com sucesso, no todo ou via *lazy query*.
2. Transforme a tabela para descobrir as informações abaixo. Em cada uma das perguntas, deixe visível o dataset resultante da transformação necessária.
    1. Qual o **tipo de reação** mais reportado na tabela?
    2. Qual o **nome da indústria** que tem **mais mortes** reportadas?
    3. Quais as **3 principais reações adversas** de pessoas entre 18-25 anos na **indústria de cosméticos**?
    4. Usando o `purrr`, escreva em uma pasta 10 tabelas diferentes com as **10 reações mais recorrentes** (uma tabela para cada reação).

In [ ]:
!pip install db-dtypes
!pip install 'google-cloud-bigquery[pandas]
!pip install google-cloud-bigquery==2.34.3

!pip install google-cloud-bigquery

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
/bin/bash: -c: line 0: unexpected EOF while looking for matching `''
/bin/bash: -c: line 1: syntax error: unexpected end of file
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
from google.cloud import bigquery
from google.oauth2 import service_account
import pandas as pd

In [ ]:
project_id = 'greenpeace-360016'
key = {"type":"service_account","project_id":"greenpeace-360016","private_key_id":"546ca86c9b77a68858f928bc8415244a93f72142","private_key":"-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQCYIehwhtnNsw4T\nd2Is/nKnoGRG9hwOj2wAZFgr1HrcIgV+jfv8CqO5jXWG3v5aItX0dffiOTR6M7iK\n3oI3ZmgPHQ3utlr9vs+L6+E2PBt7wWABIY1I2INR4cQpw9klavpo7hsEyhjMLY0v\n9IZd/jQGhgZTVWxQiaSMyydIXb20APt2ZDr4AmQ+pDoDW5jUb4RsXnEGFtNw7lv6\ntNyKQoIjjkAOfOV3zvxVVg6HevHRztPPVTZcxpuOoV/OaTs2nW0jyJcYJPSKw3a8\n91Jzz+q6Qpxn9vHxLYHd9AC6oRTEvLvPW+tKnd1BrEHUxm9fJPwWIcL8wyEHRwVa\nfOTqmeJ3AgMBAAECggEABA+nqlVXPCADCK7/rm7Odb4Y3zITtqnKesoy9lySOQ0p\nc2Lr7qMEEuB9Wi/av2I9rux1RRM83ORDkWV2OW9g6MJgTFaO4EY3OSd/WBjLrQaQ\n/wa7xE2PV+C8gVbqlY6hB6sPipWwisbOhZC0A49VKTVrts7JeO7rtxx3PH3QIh14\nCQxv/Se7vGF3TvM/yU+uXgGfvR7ouFHXkDkaoYkRqpmJ7OVQvaeL2WgIXauSPqKQ\nUWpOliZlJBoKULqQn7Qtxv6SvnXEbFwLNddG87gnk22eHgRTIhegiof529RECZRG\naqJj4DstJr8Dt8Y2EgnrXh7z74CVHVtxsJRaH9aC3QKBgQDTr8HYtJMaSQXb/CGJ\npkbKbgXYKoi+RIAuda2BZLyhpMXAq5tkCPhb5i2jNmSv/OPTcIvwGs6FofZh+Lcq\nTFB4qL8qL+GYWjG8vK2tLOgg07wBOzG3zm4aDxd6mpPZ0Oc79mEef4DK86n61woG\n7ackziX8A40T9MZJ1l0v+Co2BQKBgQC3+qhgtqKXt0Do1QctBbRMAD/nsT6mcXzm\nnUtrcRJlHc+p+iX3hg/8jedo8iErx3vfYGCmZrMyKom205diA7ehahWumn0HvXzK\noEx0xRjpWUqa74p0Yt/MKmMD2Vt2POD8muRWUtMewP+Amhjo4d9hxjed72eQ2Vp2\nAdAl4lIDSwKBgQCPRkUYbFeoEcaSdxsn9E+zkTYJcfhtdyWezG0rx8fYTBLAbiXV\nVjDq2mcHMbiCWNLGeEVutGYVkmKVOuaWDWLl16jTX+b73AgP61BNhrhyB30XBD6J\nXoGsx4wnolnxse/f6MkhAEFmm/bvXP5fZBid051n7etdl21h6mHWJsxDFQKBgB0+\nQV9GCGgXH8Eig0Z7vqHJ9F5uhOfN6C172Q4V5feemnCJK1nBvG7HW34J5o3f4meq\nGRK6Bke2DTe7QDe7IYgNpa6gLbl6PAi1HBH/dis5HyurYZWIVcxM/HoTJX+Qw65D\nLIIwHhCoWrRlSgpSRMB00khPUmXqtn9FYIF1DNYrAoGBANMViUyS3nmRY0KNZsJU\nx/yzXpujcITxYtuXWL+9IkBgVvvRJ1IKGaBpmJ7m9yyOwdWRChZWIXm7Jl79RRjJ\n7BjrPPi42DO5MO9gTCIhTDN+HM3WcJZI7LZkbgzNiOPnB6h3WDpptOMEo0JR6EUu\n8uAySj1suOIK167A0cIVYyAi\n-----END PRIVATE KEY-----\n","client_email":"bigquery@greenpeace-360016.iam.gserviceaccount.com","client_id":"101512316411573110352","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_x509_cert_url":"https://www.googleapis.com/robot/v1/metadata/x509/bigquery%40greenpeace-360016.iam.gserviceaccount.com"}

gcp_credentials = service_account.Credentials.from_service_account_info(key)

In [ ]:
query = f'''
SELECT * 
FROM fda_food.food_events
'''

food = pd.read_gbq(query, credentials=gcp_credentials,  project_id=project_id)

# food.head()

In [ ]:
food = pd.read_csv('food_events.csv')
food = food[['report_number', 'reactions', 'outcomes',
       'products_brand_name', 'products_industry_code', 'products_role',
       'products_industry_name', 'date_created', 'date_started',
       'consumer_gender', 'consumer_age', 'consumer_age_unit']]


In [ ]:
print(food.shape[0])
food.head()

91776


report_number                                          reactions  \
0  2020-CFS-001365          ABDOMINAL PAIN,DIARRHOEA,MALAISE,VOMITING   
1           157487                        NAUSEA,COUGH,ABDOMINAL PAIN   
2           173594  CHILLS,FEELING HOT,DIZZINESS,FEELING OF BODY T...   
3  2019-CFS-011609  DIARRHOEA,DYSPNOEA,OROPHARYNGEAL PAIN,PYREXIA,...   
4           167726                                          DIARRHOEA   

                              outcomes  \
0                    Other Seriousness   
1  Patient Visited Healthcare Provider   
2                        Other Outcome   
3                    Other Seriousness   
4                        Other Outcome   

                       products_brand_name products_industry_code  \
0         BAKER'S CORNER ALL PURPOSE FLOUR                      2   
1          KROGER ENRICHED LONG GRAIN RICE                      2   
2  UNCLE BEN'S INFUSED RICE MEXICAN FLAVOR                      2   
3                     PREMIUM PATENT FLOUR                      2   
4  CANILLA EXTRA LONG GRAIN, ENRICHED RICE                      2   

  products_role                products_industry_name date_created  \
0       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2020-02-05   
1       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2012-10-09   
2       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2014-01-31   
3       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2019-11-23   
4       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2013-07-12   

  date_started consumer_gender  consumer_age consumer_age_unit  
0          NaN             NaN           NaN               NaN  
1          NaN             NaN           NaN               NaN  
2   2013-12-03             NaN           NaN               NaN  
3          NaN             NaN           NaN               NaN  
4          NaN             NaN           NaN               NaN

# Qual o tipo de reação mais reportado na tabela?

In [ ]:


reactions = food['reactions'].value_counts()
reactions.head()

OVARIAN CANCER    10157
CHOKING            1759
DIARRHOEA          1305
ALOPECIA           1056
VOMITING           1051
Name: reactions, dtype: int64

A reação mais repostada na tabela foi `OVARIAN CANCER`, com `10157` ocorrências

# Qual o nome da indústria que tem mais mortes reportadas?

In [ ]:
outcomes = food['outcomes'].value_counts()
outcomes.head()

Medically Important                                        24771
Other Outcome                                              23528
Patient Visited Healthcare Provider,Medically Important     6357
Patient Visited Healthcare Provider                         4672
Hospitalization                                             4130
Name: outcomes, dtype: int64

In [ ]:
death = food[food['outcomes'] == 'Death']
print(death.shape[0])
death.head()

4077


report_number        reactions outcomes  \
328          70500            DEATH    Death   
1823        189865            DEATH    Death   
4433        209772            DEATH    Death   
4565        209774            DEATH    Death   
5523         71664  DIARRHOEA,DEATH    Death   

                    products_brand_name products_industry_code products_role  \
328                TRE'S BON CHEESECAKE                      3       SUSPECT   
1823  NATURE'S VALLEY HONEY WHEAT BREAD                      3       SUSPECT   
4433                        EXEMPTION 4                     12       SUSPECT   
4565                        EXEMPTION 4                     12       SUSPECT   
5523                             OYSTER                     16       SUSPECT   

           products_industry_name date_created date_started consumer_gender  \
328   Bakery Prod/Dough/Mix/Icing   2004-07-16   2004-04-26             NaN   
1823  Bakery Prod/Dough/Mix/Icing   2015-09-23   2015-09-20             NaN   
4433           Cheese/Cheese Prod   2017-03-30          NaN             NaN   
4565           Cheese/Cheese Prod   2017-03-30          NaN             NaN   
5523         Fishery/Seafood Prod   2004-08-26          NaN             NaN   

      consumer_age consumer_age_unit  
328            NaN               NaN  
1823           NaN               NaN  
4433           NaN               NaN  
4565           NaN               NaN  
5523           NaN               NaN

In [ ]:
industry_name = food['products_industry_name'].value_counts()
industry_name.head(1)

Vit/Min/Prot/Unconv Diet(Human/Animal)    25401
Name: products_industry_name, dtype: int64

O nome da industria que tem mais mortes reportadas é `Vit/Min/Prot/Unconv Diet(Human/Animal)`, com `25401` ocorrências

# 3 - Quais as 3 principais reações adversas de pessoas entre 18-25 anos na indústria de cosméticos?

In [ ]:
# SELECT * FROM `bigquery-public-data.fda_food.food_events`
# WHERE consumer_age BETWEEN 18 AND 25

peoples_25 = food[(food['consumer_age']>= 18) &(food['consumer_age']<= 25)]
print(peoples_25.shape[0])

peoples_25.head()

2932


report_number                                          reactions  \
48965  2018-CFS-011979      ABDOMINAL PAIN UPPER,DIARRHOEA,ABDOMINAL PAIN   
48966           142925                                      RASH,PRURITUS   
48967           116105                         VOMITING,MALAISE,DIARRHOEA   
48968           180567                                             NAUSEA   
48969           210557  DIARRHOEA,ABDOMINAL PAIN,VOMITING,FEELING OF B...   

                                       outcomes  \
48965                       Medically Important   
48966                             Other Outcome   
48967                             Other Outcome   
48968                             Other Outcome   
48969  Patient Visited ER,Required Intervention   

                                    products_brand_name  \
48965                                   LONG GRAIN RICE   
48966  MARCUS BROTHERS EL MEXICANO PREMIUM QUALITY RICE   
48967           RICHFOOD ENRICHED LONG GRAIN WHITE RICE   
48968              HOSTESS DONNETTES WHITE POWDER SUGAR   
48969         HOSTESS HOLIDAY WHITE PEPPERMINT TWINKIES   

      products_industry_code products_role  \
48965                      2       SUSPECT   
48966                      2       SUSPECT   
48967                      2       SUSPECT   
48968                      3       SUSPECT   
48969                      3       SUSPECT   

                     products_industry_name date_created date_started  \
48965  Whole Grain/Milled Grain Prod/Starch   2018-09-29   2018-08-07   
48966  Whole Grain/Milled Grain Prod/Starch   2011-08-30          NaN   
48967  Whole Grain/Milled Grain Prod/Starch   2009-07-28          NaN   
48968           Bakery Prod/Dough/Mix/Icing   2014-11-04          NaN   
48969           Bakery Prod/Dough/Mix/Icing   2017-04-18          NaN   

      consumer_gender  consumer_age consumer_age_unit  
48965               F          18.0           year(s)  
48966               F          18.0           year(s)  
48967               M          18.0           year(s)  
48968               M          18.0           year(s)  
48969               M          18.0           year(s)

In [ ]:
peoples_25_reactions = peoples_25['reactions'].value_counts()
peoples_25_reactions.head(3)

OVARIAN CANCER    55
ALOPECIA          44
VOMITING          42
Name: reactions, dtype: int64

As 3 principais reações adversas de pessoas entre 18-25 anos na indústria de cosméticos são `OVARIAN CANCER`, `ALOPECIA` e `VOMITING`, respectivamente com `55`, `44` e `42` ocorrências

# Escreva em uma pasta 10 tabelas diferentes com as 10 reações mais recorrentes (uma tabela para cada reação).

In [ ]:
reactions = food['reactions'].value_counts()
reactions.head(10)

OVARIAN CANCER          10157
CHOKING                  1759
DIARRHOEA                1305
ALOPECIA                 1056
VOMITING                 1051
DEATH,INJURY              851
INJURY,DEATH              834
NAUSEA                    817
MALAISE                   717
OVARIAN CANCER,DEATH      694
Name: reactions, dtype: int64

In [ ]:
ovarian_cancer_death = food[food['reactions']=='OVARIAN CANCER,DEATH']
print(ovarian_cancer_death.shape[0])
ovarian_cancer_death.to_excel('10_ovarian_cancer_death.xlsx')
ovarian_cancer_death.head(1)

694


report_number             reactions                   outcomes  \
18189  2018-CFS-010763  OVARIAN CANCER,DEATH  Death,Medically Important   

      products_brand_name products_industry_code products_role  \
18189         EXEMPTION 4                     53       SUSPECT   

      products_industry_name date_created date_started consumer_gender  \
18189              Cosmetics   2018-08-28   2010-01-01             NaN   

       consumer_age consumer_age_unit  
18189           NaN               NaN

In [ ]:
malaise = food[food['reactions']=='MALAISE']
print(malaise.shape[0])
malaise.to_excel('9_malaise.xlsx')
malaise.head(1)

717


report_number reactions                                           outcomes  \
59         66460   MALAISE  Patient Visited Healthcare Provider,Patient Vi...   

                   products_brand_name products_industry_code products_role  \
59  GREAT VALUE QUICK GRITS (ENRICHED)                      2       SUSPECT   

                  products_industry_name date_created date_started  \
59  Whole Grain/Milled Grain Prod/Starch   2004-02-07          NaN   

   consumer_gender  consumer_age consumer_age_unit  
59             NaN           NaN               NaN

In [ ]:
nausea = food[food['reactions']=='NAUSEA']
print(nausea.shape[0])
nausea.to_excel('8_nausea.xlsx')
nausea.head(1)

817


report_number reactions       outcomes  \
68         77285    NAUSEA  Other Outcome   

                      products_brand_name products_industry_code  \
68  UNCLE BEN'S READY RICE (MICROWAVABLE)                      2   

   products_role                products_industry_name date_created  \
68       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2005-04-05   

   date_started consumer_gender  consumer_age consumer_age_unit  
68   2005-03-25             NaN           NaN               NaN

In [ ]:
injury_death = food[food['reactions']=='INJURY,DEATH']
print(injury_death.shape[0])
injury_death.to_excel('7_injury_death.xlsx')
injury_death.head(1)

834


report_number     reactions outcomes products_brand_name  \
18185  2019-CFS-004347  INJURY,DEATH    Death         EXEMPTION 4   

      products_industry_code products_role products_industry_name  \
18185                     53       SUSPECT              Cosmetics   

      date_created date_started consumer_gender  consumer_age  \
18185   2019-04-30          NaN             NaN           NaN   

      consumer_age_unit  
18185               NaN

In [ ]:
death_injury = food[food['reactions']=='DEATH,INJURY']
print(death_injury.shape[0])
death_injury.to_excel('6_death_injury.xlsx')
death_injury.head(1)

851


report_number     reactions outcomes products_brand_name  \
18323  2018-CFS-010638  DEATH,INJURY    Death         EXEMPTION 4   

      products_industry_code products_role products_industry_name  \
18323                     53       SUSPECT              Cosmetics   

      date_created date_started consumer_gender  consumer_age  \
18323   2018-08-24          NaN             NaN           NaN   

      consumer_age_unit  
18323               NaN

In [ ]:
# repetir padrão para todas as outras 9 reações

vomiting = food[food['reactions']=='VOMITING']
print(vomiting.shape[0])
vomiting.to_excel('5_vomiting.xlsx')
vomiting.head(1)

1051


report_number reactions    outcomes products_brand_name  \
52  2017-CFS-001779  VOMITING  Disability   SILA BASMATI RICE   

   products_industry_code products_role                products_industry_name  \
52                      2       SUSPECT  Whole Grain/Milled Grain Prod/Starch   

   date_created date_started consumer_gender  consumer_age consumer_age_unit  
52   2017-11-16   2017-08-23             NaN           NaN               NaN

In [ ]:
# repetir padrão para todas as outras 9 reações

alopecia = food[food['reactions']=='ALOPECIA']
print(alopecia.shape[0])
alopecia.to_excel('4_alopecia.xlsx')
alopecia.head(1)

1056


report_number reactions                        outcomes  \
18197        207303  ALOPECIA  Disability,Medically Important   

      products_brand_name products_industry_code products_role  \
18197    WEN BY CHAZ DEAN                     53       SUSPECT   

      products_industry_name date_created date_started consumer_gender  \
18197              Cosmetics   2017-02-08   2016-12-06             NaN   

       consumer_age consumer_age_unit  
18197           NaN               NaN

In [ ]:
# repetir padrão para todas as outras 9 reações

diarrhoea = food[food['reactions']=='DIARRHOEA']
print(diarrhoea.shape[0])
diarrhoea.to_excel('3_diarrhoea.xlsx')
diarrhoea.head(1)

1305


report_number  reactions       outcomes  \
4        167726  DIARRHOEA  Other Outcome   

                       products_brand_name products_industry_code  \
4  CANILLA EXTRA LONG GRAIN, ENRICHED RICE                      2   

  products_role                products_industry_name date_created  \
4       SUSPECT  Whole Grain/Milled Grain Prod/Starch   2013-07-12   

  date_started consumer_gender  consumer_age consumer_age_unit  
4          NaN             NaN           NaN               NaN

In [ ]:
# repetir padrão para todas as outras 9 reações

choking = food[food['reactions']=='CHOKING']
print(choking.shape[0])
choking.to_excel('2_choking.xlsx')
choking.head(1)

1759


report_number reactions       outcomes          products_brand_name  \
269         73317   CHOKING  Other Outcome  SUNSHINE CHEEZE-IT CRACKERS   

    products_industry_code products_role       products_industry_name  \
269                      3       SUSPECT  Bakery Prod/Dough/Mix/Icing   

    date_created date_started consumer_gender  consumer_age consumer_age_unit  
269   2004-10-28   2004-10-09             NaN           NaN               NaN

In [ ]:
# repetir padrão para todas as outras 9 reações

ovarian_cancer = food[food['reactions']=='OVARIAN CANCER']
print(ovarian_cancer.shape[0])
ovarian_cancer.to_excel('1_ovarian_cancer.xlsx')
ovarian_cancer.head(1)

10157


report_number       reactions             outcomes products_brand_name  \
18166        185377  OVARIAN CANCER  Medically Important         EXEMPTION 4   

      products_industry_code products_role products_industry_name  \
18166                     53       SUSPECT              Cosmetics   

      date_created date_started consumer_gender  consumer_age  \
18166   2015-04-27          NaN             NaN           NaN   

      consumer_age_unit  
18166               NaN

# update data bigquery

In [ ]:
project_id = 'greenpeace-360016'
key = {"type":"service_account","project_id":"greenpeace-360016","private_key_id":"546ca86c9b77a68858f928bc8415244a93f72142","private_key":"-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQCYIehwhtnNsw4T\nd2Is/nKnoGRG9hwOj2wAZFgr1HrcIgV+jfv8CqO5jXWG3v5aItX0dffiOTR6M7iK\n3oI3ZmgPHQ3utlr9vs+L6+E2PBt7wWABIY1I2INR4cQpw9klavpo7hsEyhjMLY0v\n9IZd/jQGhgZTVWxQiaSMyydIXb20APt2ZDr4AmQ+pDoDW5jUb4RsXnEGFtNw7lv6\ntNyKQoIjjkAOfOV3zvxVVg6HevHRztPPVTZcxpuOoV/OaTs2nW0jyJcYJPSKw3a8\n91Jzz+q6Qpxn9vHxLYHd9AC6oRTEvLvPW+tKnd1BrEHUxm9fJPwWIcL8wyEHRwVa\nfOTqmeJ3AgMBAAECggEABA+nqlVXPCADCK7/rm7Odb4Y3zITtqnKesoy9lySOQ0p\nc2Lr7qMEEuB9Wi/av2I9rux1RRM83ORDkWV2OW9g6MJgTFaO4EY3OSd/WBjLrQaQ\n/wa7xE2PV+C8gVbqlY6hB6sPipWwisbOhZC0A49VKTVrts7JeO7rtxx3PH3QIh14\nCQxv/Se7vGF3TvM/yU+uXgGfvR7ouFHXkDkaoYkRqpmJ7OVQvaeL2WgIXauSPqKQ\nUWpOliZlJBoKULqQn7Qtxv6SvnXEbFwLNddG87gnk22eHgRTIhegiof529RECZRG\naqJj4DstJr8Dt8Y2EgnrXh7z74CVHVtxsJRaH9aC3QKBgQDTr8HYtJMaSQXb/CGJ\npkbKbgXYKoi+RIAuda2BZLyhpMXAq5tkCPhb5i2jNmSv/OPTcIvwGs6FofZh+Lcq\nTFB4qL8qL+GYWjG8vK2tLOgg07wBOzG3zm4aDxd6mpPZ0Oc79mEef4DK86n61woG\n7ackziX8A40T9MZJ1l0v+Co2BQKBgQC3+qhgtqKXt0Do1QctBbRMAD/nsT6mcXzm\nnUtrcRJlHc+p+iX3hg/8jedo8iErx3vfYGCmZrMyKom205diA7ehahWumn0HvXzK\noEx0xRjpWUqa74p0Yt/MKmMD2Vt2POD8muRWUtMewP+Amhjo4d9hxjed72eQ2Vp2\nAdAl4lIDSwKBgQCPRkUYbFeoEcaSdxsn9E+zkTYJcfhtdyWezG0rx8fYTBLAbiXV\nVjDq2mcHMbiCWNLGeEVutGYVkmKVOuaWDWLl16jTX+b73AgP61BNhrhyB30XBD6J\nXoGsx4wnolnxse/f6MkhAEFmm/bvXP5fZBid051n7etdl21h6mHWJsxDFQKBgB0+\nQV9GCGgXH8Eig0Z7vqHJ9F5uhOfN6C172Q4V5feemnCJK1nBvG7HW34J5o3f4meq\nGRK6Bke2DTe7QDe7IYgNpa6gLbl6PAi1HBH/dis5HyurYZWIVcxM/HoTJX+Qw65D\nLIIwHhCoWrRlSgpSRMB00khPUmXqtn9FYIF1DNYrAoGBANMViUyS3nmRY0KNZsJU\nx/yzXpujcITxYtuXWL+9IkBgVvvRJ1IKGaBpmJ7m9yyOwdWRChZWIXm7Jl79RRjJ\n7BjrPPi42DO5MO9gTCIhTDN+HM3WcJZI7LZkbgzNiOPnB6h3WDpptOMEo0JR6EUu\n8uAySj1suOIK167A0cIVYyAi\n-----END PRIVATE KEY-----\n","client_email":"bigquery@greenpeace-360016.iam.gserviceaccount.com","client_id":"101512316411573110352","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_x509_cert_url":"https://www.googleapis.com/robot/v1/metadata/x509/bigquery%40greenpeace-360016.iam.gserviceaccount.com"}



In [ ]:
from google.cloud import bigquery
from google.oauth2 import service_account

gcp_credentials = service_account.Credentials.from_service_account_info(key)

In [ ]:
food.to_gbq(destination_table='fda_food.food_events', project_id=project_id,
                            credentials=gcp_credentials, if_exists='replace')

1it [00:07,  7.84s/it]
